### DISCUSSION TITLE
esp32s3 ili9341. Display only weakly blinking.

### DISCUSSION THREAD

**Discussion by fedusiv** [Bodmer/TFT_eSPI#2125] [General]
Hi!
I have issue with using ili9341. After all manipulation and setup display only blinks weakly each time tft.fillRect is called.

I have esp32s3 N16R8
![image](https://user-images.githubusercontent.com/22212232/198972760-10af13d4-572f-4da6-a72e-2b292110c5bc.png)

ili9341 display.
![image](https://user-images.githubusercontent.com/22212232/198973034-5a651b89-6f07-4046-a667-c938d86d0337.png)

I connected spi as mentioned alternative for esp32s3 SPI2 or FSPI.
`#define TFT_CS   10 //     10 or 34 (FSPI CS0)`
`#define TFT_MOSI 11 //     11 or 35 (FSPI D)`
`#define TFT_SCLK 12//     12 or 36 (FSPI CLK)`
`#define TFT_MISO 13 //     13 or 37 (FSPI Q)`

LED pin or backlight is connected to 3.3V. Display and esp are powered from 3.3V

For code I used ellipse example https://github.com/Bodmer/TFT_eSPI/blob/master/examples/320%20x%20240/TFT_Ellipse/TFT_Ellipse.ino

Configs. 
In User_Setup_Select.h I commented 
`#include <User_Setup.h> `
and uncommented
`#include <User_Setups/Setup70b_ESP32_S3_ILI9341.h>`

In config file I changed PIN definition as mentioned above.
Added RGB instead of BGR and specify height and width (320x240)

But display only blinks. No picture, no drawing, nothing. Only looks like it on and off backlight with 2000 ms delay.
No runtime errors.

Please help me to explain what is issue here.

**Comment by fedusiv** [comment]
Display has started to showing something, once I changed SPi freq to 5 MHz.
But it works slooow. Correctly works only fillRect, but only with specific colors. What it can be?

**Reply by SynAckShaman** [reply]
The issue is your not using circuitpython.

this is the firmware for you board. 
https://circuitpython.org/board/yd_esp32_s3_n16r8/

here is the code for your board and display

```
"""
This test will initialize the display using displayio and draw a solid green
background, a smaller purple rectangle, and some yellow text. All drawing is done
using native displayio modules.

Pinouts are for the 2.4" TFT FeatherWing or Breakout with a Feather M4 or M0.
"""
import board
import busio
import terminalio
import displayio
from adafruit_display_text import label
import adafruit_ili9341

# Support both 8.x.x and 9.x.x. Change when 8.x.x is discontinued as a stable release.
try:
    from fourwire import FourWire
except ImportError:
    from displayio import FourWire

# Release any resources currently in use for the displays
displayio.release_displays()

spi = busio.SPI(board.GPIO12, MISO=board.GPIO13, MOSI=board.GPIO11) 
tft_cs = board.GPIO10
tft_dc = board.GPIO7
tft_rst = board.GPIO6

display_bus = FourWire(spi, command=tft_dc, chip_select=tft_cs, reset=tft_rst)
display = adafruit_ili9341.ILI9341(display_bus, width=320, height=240)

# Make the display context
splash = displayio.Group()
display.root_group = splash

# Draw a green background
color_bitmap = displayio.Bitmap(320, 240, 1)
color_palette = displayio.Palette(1)
color_palette[0] = 0x00FF00  # Bright Green

bg_sprite = displayio.TileGrid(color_bitmap, pixel_shader=color_palette, x=0, y=0)

splash.append(bg_sprite)

# Draw a smaller inner rectangle
inner_bitmap = displayio.Bitmap(280, 200, 1)
inner_palette = displayio.Palette(1)
inner_palette[0] = 0xAA0088  # Purple
inner_sprite = displayio.TileGrid(inner_bitmap, pixel_shader=inner_palette, x=20, y=20)
splash.append(inner_sprite)

# Draw a label
text_group = displayio.Group(scale=3, x=57, y=120)
text = "Hello World!"
text_area = label.Label(terminalio.FONT, text=text, color=0xFFFF00)
text_group.append(text_area)  # Subgroup for text scaling
splash.append(text_group)

while True:
    pass
```

**Comment by FerGT50** [comment]
> I connected spi as mentioned alternative for esp32s3 SPI2 or FSPI. `#define TFT_CS 10 // 10 or 34 (FSPI CS0)` `#define TFT_MOSI 11 // 11 or 35 (FSPI D)` `#define TFT_SCLK 12// 12 or 36 (FSPI CLK)` `#define TFT_MISO 13 // 13 or 37 (FSPI Q)`

This seems to use the "SPI2" controller.

In your Setup70b_ESP32_S3_ILI9341.h, you may try adding:

`#define TFT_SPI_PORT 2
`
